### PR TITLE
scrub aggregator names to eliminate discovery

### PR DESCRIPTION
**Pull request by deads2k**
Cleanup old uses of `discovery`.  Also removes the legacy functionality.

@kubernetes/sig-api-machinery-misc @sttts

### CODE REVIEW

**Comment by smarterclayton** [comment]
Changes LGTM (I didn't see anything that surprised me).  No implications to any of the deployer tools?

**Comment by deads2k** [comment]
> Changes LGTM (I didn't see anything that surprised me). No implications to any of the deployer tools?

No.  They'd use the binary name.

**Comment by deads2k** [comment]
@k8s-bot gce etcd3 e2e test this

File: test/integration/kubeaggregator/aggregator_test.go [hunk 1]
```diff
@@ -14,15 +14,14 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package discoverysummarizer
+package kubeaggregator
```
Comment (sttts) [inline test/integration/kubeaggregator/aggregator_test.go]: general question: how kube specific is all this? Not at all? Let's call it in a neutral fashion.

Comment (deads2k) [inline test/integration/kubeaggregator/aggregator_test.go]: > general question: how kube specific is all this? Not at all? Let's call it in a neutral fashion.

It seems to test to make sure the aggregator can start at all.  It predated the aggregator and started with `kubernetes-discovery`.

**Comment by deads2k** [comment]
@apelisse The bot auto-applies "approved" since I'm in the lists, but then doesn't auto-reapply on a new push.

**Comment by apelisse** [comment]
@deads2k I'll push the code that makes the approval label sticky (i.e. approval stays after code push).

**Pull request merged**